tests/logger/logger.test.ts

```diff
@@ -0,0 +1,60 @@
+import Logger, { SuggestedLogObject } from '../../src/logger/logger';
+
+describe('log message', () => {
+  const logFunction = jest.fn();
```

Comment:
The `logFunction` created in describe block is now being shared in all the 3 tests. So the invocations that you are verifying are going to accumulate. I.e. in the last test, invocations from all the other tests are available.

This might cause the tests to influence each other. You have two options how to avoid it:
1. instantiate the mocked `logFunction` in each `test` block
2. add `afterEach(() => {logFunction.mockReset();})` to reset the mock after each test

Reply:
didn't thought about that, added...
`afterEach(() => {
    jest.restoreAllMocks();
  });`

Reply:
Is restoring what you really wanted?

Reading the docs https://jestjs.io/docs/jest-object#jestrestoreallmocks it seems to restore the mocked behavior only for `spyOn`.

The `resetAllMocks` sounds more like what we need here https://jestjs.io/docs/jest-object#jestresetallmocks

Reply:
nerd..., who reads documentation when he does not have a problem to solve? :D